### PR TITLE
Move Azure Tables into sdk/tables/azure-data-tables

### PR DESCRIPTION
**Pull request by conniey**
* Moves Azure Tables client library into `sdk/tables/azure-data-tables`. Follows the structure used by .NET (https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/tables/Azure.Data.Tables).
* Alphabetises jacoco-test-coverage/pom.xml dependencies. :(

If this PR is merged, this one can be closed:  https://github.com/Azure/azure-sdk-for-java/pull/11661

### CODE REVIEW

File: eng/jacoco-test-coverage/pom.xml [hunk 1]
```diff
@@ -91,30 +91,35 @@
...
     <dependency>
       <groupId>com.azure</groupId>
       <artifactId>azure-data-appconfiguration</artifactId>
       <version>1.2.0-beta.1</version> <!-- {x-version-update;com.azure:azure-data-appconfiguration;current} -->
     </dependency>
     <dependency>
       <groupId>com.azure</groupId>
-      <artifactId>azure-identity</artifactId>
-      <version>1.1.0-beta.5</version> <!-- {x-version-update;com.azure:azure-identity;current} -->
+      <artifactId>azure-data-schemaregistry</artifactId>
```
Comment (conniey) [inline eng/jacoco-test-coverage/pom.xml:106]: Most of these are alphabetise changes where I cut and paste the dependencies into the right alphbetical order. The one that matters is line 116 - 117.

File: sdk/tables/azure-data-tables/pom.xml [hunk 2]
```diff
@@ -12,11 +12,10 @@ Licensed under the MIT License.
...
 
   <groupId>com.azure</groupId>
-  <artifactId>azure-cosmos-table</artifactId>
-  <version>1.0.0-beta.1</version> <!-- {x-version-update;com.azure:azure-cosmos-table;current} -->
+  <artifactId>azure-data-tables</artifactId>
+  <version>1.0.0-beta.1</version> <!-- {x-version-update;com.azure:azure-data-tables;current} -->
   <name>Microsoft Azure SDK for Azure Table</name>
-  <description>This package contains the Microsoft Azure Cosmos Table client library.</description>
-  <packaging>jar</packaging>
+  <description>This package contains the Microsoft Azure Storage Table client library.</description>
```
Comment (srnagar) [inline sdk/tables/azure-data-tables/pom.xml:18]: Should the group name be `data` or `storage`? The description says `Azure Storage Table`.

File: eng/jacoco-test-coverage/pom.xml (hunk 1 above)
Comment (srnagar) [reply eng/jacoco-test-coverage/pom.xml:106]: This is great! Thank you!

**Comment by mitchdenny** [comment]
@conniey this PR will also need to remove the module from additional modules in ```sdk/cosmos/ci.yml```.

**Pull request merged**